sysfs/net_class.go

```diff
@@ -117,14 +118,35 @@ func (fs FS) NetClass() (NetClass, error) {
 	return netClass, nil
 }
 
+// isFatalError returns true if the error should terminate attribute reading.
+func isFatalError(err error) bool {
+	var errno syscall.Errno
+
+	if os.IsNotExist(err) {
+		return false
+	} else if os.IsPermission(err) {
+		return false
+	} else if err.Error() == "operation not supported" {
+		return false
+	} else if errors.Is(err, os.ErrInvalid) {
+		return false
+	} else if errors.As(err, &errno) && (errno == syscall.EINVAL) {
+		return false
+	}
+	return true
+}
+
 // ParseNetClassAttribute parses a given file in /sys/class/net/<iface>
 // and sets the value in a given NetClassIface object if the value was readable.
-// It returns an error if the file cannot be read.
+// It returns an error if the file cannot be read and the error is fatal.
 func ParseNetClassAttribute(devicePath, attrName string, interfaceClass *NetClassIface) error {
 	attrPath := filepath.Join(devicePath, attrName)
 	value, err := util.SysReadFile(attrPath)
 	if err != nil {
-		return fmt.Errorf("failed to read file %q: %w", attrPath, err)
+		if isFatalError(err) {
+			return fmt.Errorf("failed to read file %q: %w", attrPath, err)
+		}
+		return nil
```

Comment:
Hmm, what errors are we going to ignore by doing this?

Reply:
@SuperQ No more errors than we ignored before... It uses the same logic, but now additionally ignores the syscall.Errno(EINVAL)